// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc SayHelloStreamReply (HelloRequest) returns (stream HelloReply) {}

  rpc SayHelloBidiStream (stream HelloRequest) returns (stream HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

// client methods to master server
service clientMethods {
  rpc Setup(clientRequest) returns (stream clientResponse) {}
  rpc Get(clientRequest) returns (stream clientResponse) {}
  rpc Put(clientRequest) returns (stream clientResponse) {}
  rpc Del(clientRequest) returns (stream clientResponse) {}
}

message clientRequest {
  string method = 1;
  string key = 2;
  string value = 3;
}

message clientResponse {
  string response = 1;
  string state = 2;
  repeated int32 port = 3;
}


// worker register
service workerRegister {

}

message Register {
  string request = 1;
}

